lib/mutant/registry.rb

```diff
@@ -0,0 +1,44 @@
+module Mutant
+  # Registry for mapping AST types to classes
+  class Registry
+    include Concord.new(:contents)
+
+    # Initialize object
+    #
+    # @return [undefined]
+    def initialize
+      super({})
+    end
+
+    # Raised when the type is an invalid type
+    RegistryError = Class.new(TypeError)
+
+    # Register class for AST node class
+    #
+    # @param [Symbol] type
+    # @param [Class] class
+    #
+    # @return [self]
+    def register(type, klass)
+      fail RegistryError, "Invalid type registration: #{type.inspect}" unless AST::Types::ALL.include?(type)
```

Comment:
Sure I can do that. I thought this project preferred 120 chars but I can make the change